lms/djangoapps/grades/tests/test_signals.py

```diff
@@ -251,3 +256,37 @@ def test_disconnect_manager_bad_arg(self):
         with self.assertRaises(ValueError):
             with disconnect_submissions_signal_receiver(PROBLEM_RAW_SCORE_CHANGED):
                 pass
+
+
+@ddt.ddt
+class UserPartitionChangeHandlerTest(TestCase):
+    """
+    Tests to ensure that sending signals related to user
+    partition changes triggers a recalculation of the affected
+    user's course grade.
+    """
+    def setUp(self):
+        super(UserPartitionChangeHandlerTest, self).setUp()
+        self.user = UserFactory()
+        self.course_key = CourseLocator("test_org", "test_course_num", "test_run")
+
+    @patch('lms.djangoapps.grades.signals.handlers.CourseGradeFactory.update')
+    @patch('lms.djangoapps.grades.signals.handlers.CourseGradeFactory.read')
+    @ddt.data(*itertools.product((COHORT_MEMBERSHIP_UPDATED, ENROLLMENT_TRACK_UPDATED), (True, False), (True, False)))
+    @ddt.unpack
+    def test_enrollment_signal(self, signal, write_only_if_engaged, has_grade, read_mock, update_mock):
```

Comment:
If/once you remove the additional `recalculate_course_grade_on_user_partition_change` method, this test could be much simpler.

Should this test go in test_integrations.py since this is integrated with other domain concepts?

Reply:
I wasn't quite sure where to put this one. I think the other "integration" tests actually execute the entire workflow related to those integrations- they don't mock out the other services, but actually call them. Here, I'm just testing that given that a signal is sent, this code reacts the way we expect it to. That felt more like a unit test than integration to me, but I'm happy to move it.